cpp/src/arrow/scalar_test.cc

```diff
@@ -37,14 +37,52 @@ namespace arrow {
 using internal::checked_cast;
 using internal::checked_pointer_cast;
 
+std::shared_ptr<Scalar> CheckMakeNullScalar(const std::shared_ptr<DataType>& type) {
+  const auto scalar = MakeNullScalar(type);
+  ARROW_EXPECT_OK(scalar->Validate());
+  ARROW_EXPECT_OK(scalar->ValidateFull());
+  AssertTypeEqual(*type, *scalar->type);
+  EXPECT_FALSE(scalar->is_valid);
+  return scalar;
+}
+
+template <typename... MakeScalarArgs>
+void AssertMakeScalar(const Scalar& expected, MakeScalarArgs&&... args) {
+  ASSERT_OK_AND_ASSIGN(auto scalar, MakeScalar(std::forward<MakeScalarArgs>(args)...));
+  ASSERT_OK(scalar->Validate());
+  ASSERT_OK(scalar->ValidateFull());
+  AssertScalarsEqual(expected, *scalar, /*verbose=*/true);
+}
+
+void AssertParseScalar(const std::shared_ptr<DataType>& type, const util::string_view& s,
```

Comment:
Now that compute::Cast supports scalars for all types, is it worth keeping Scalar::Parse and Scalar::CastTo?

Reply:
I have no idea. Can you open a JIRA for that?

Reply:
will do

Reply:
https://issues.apache.org/jira/browse/ARROW-9006